### PR TITLE
Enables GQA support in the prefix prefill kernels

### PR DESCRIPTION
**Pull request by sighingnow**

### CODE REVIEW

**Comment by sighingnow** [comment]
The failure in CI's "Model Test" shouldn't be caused by this pull request and I have noticed the same failure in other PR as well as main.

**Review by WoosukKwon — APPROVED** [review]
@sighingnow Awesome! Thanks for submitting the PR! Left a minor comment on a variable name.

File: vllm/model_executor/layers/triton_kernel/prefix_prefill.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def _fwd_kernel(
         stride_v_cache_h,
         stride_v_cache_d,
         stride_v_cache_bl,
+        kv_group_num: int,
```
[the remark this answers is not in the document]
Comment (sighingnow) [reply vllm/model_executor/layers/triton_kernel/prefix_prefill.py]: Fixed. Renamed to `num_queries_per_kv` (the same with which is used in attention.py).

**Comment by WoosukKwon** [comment]
@sighingnow Thanks for the fix! I will merge the PR once it passes the CI tests.

**Comment by sighingnow** [comment]
> @sighingnow Thanks for the fix! I will merge the PR once it passes the CI tests.

Thank you!

**Comment by sighingnow** [comment]
> @sighingnow Thanks for the fix! I will merge the PR once it passes the CI tests.

Hi @WoosukKwon, CI turns green now. (Just a polite reminding).

File: tests/kernels/test_prefix_prefill.py [hunk 2]
```diff
@@ -17,12 +18,14 @@
 
 
 @pytest.mark.parametrize("num_heads", NUM_HEADS)
+@pytest.mark.parametrize("num_queries_per_kv", NUM_HEADS)
```
Comment (UranusSeven) [inline tests/kernels/test_prefix_prefill.py:21]: Should this parameter be NUM_QUERIES_PER_KV?

Comment (sighingnow) [reply tests/kernels/test_prefix_prefill.py:21]: Fixed in #3246.

**Pull request merged**